sdcm/scylla_bench_thread.py

```diff
@@ -142,13 +142,74 @@ def __init__(self, stress_cmd, loader_set, timeout, node_list=None, round_robin=
         self.hdr_tags = ["co-fixed"]
 
+    def get_datacenter_name_for_loader(self, loader):
+        datacenter_name_per_region = self.loader_set.get_datacenter_name_per_region(db_nodes=self.node_list)
+        datacenter_cmd = ""
+        if loader_dc := datacenter_name_per_region.get(loader.region):
+            datacenter_cmd = f" -datacenter={loader_dc} "
+        else:
+            LOGGER.error("Not found datacenter for loader region '%s'. Datacenter per loader dict: %s",
+                         loader.region, datacenter_name_per_region)
+        return datacenter_cmd
+
+    def adjust_cmd_node_option(self, stress_cmd, loader, cmd_runner):
+        """
+            Adjusts the stress command to include node and rack options based on the current configuration.
+            This method modifies the provided `stress_cmd` string to specify which database nodes and racks
+            should be targeted by the stress tool. It appends node IPs and, if applicable, rack and datacenter
+            information to the command. The adjustments depend on the loader's region, rack, and the test
+            configuration parameters.
+            Args:
+                stress_cmd (str): The initial stress command to be adjusted.
+                loader: The loader node object for which the command is being constructed.
+                cmd_runner: The object used to run commands on the loader.
+            Returns:
+                str: The adjusted stress command string with node and rack options appended as needed.
+        """
+        LOGGER.debug("Start adjusting stress command to use nodes: %s", stress_cmd)
+        dc_options_cmd = ""
+        if self.params.get("rack_aware_loader"):
+            LOGGER.debug("Rack-aware loader is enabled, trying to pin stress command to rack")
+            # if there are multiple rack/AZs configured, we'll try to configue s-b to pin to them
+            rack_names = self.loader_set.get_rack_names_per_datacenter_and_rack_idx(db_nodes=self.node_list)
+            by_region_rack_names = self.loader_set.get_rack_names_per_datacenter_from_rack_mapping(rack_names)
+            if any(len(racks) > 1 for racks in by_region_rack_names.values()):
+                if not (loader_rack := rack_names.get((str(loader.region), str(loader.rack)))):
+                    # NOTE: fail fast if we cannot find proper rack value when rack-awareness is enabled
+                    raise ValueError(
+                        f"Rack name not found for loader {loader} in region {loader.region} and rack {loader.rack}")
+
+                dc_options_cmd = f"-rack={loader_rack} "
+
+                if not (loader_dc := self.get_datacenter_name_for_loader(loader)):
+                    # NOTE: fail fast if we cannot find datacenter name
+                    raise ValueError(
+                        f"Datacenter name not found for loader {loader} in region {loader.region} and rack {loader.rack}")
```

Comment:
This part is duplication of the lines 192-197.
We can just remove it and update the line 192 (condition) to be something like following:
```
if self.loader_set.test_config.MULTI_REGION or "-rack=" in dc_options_cmd:
```

Reply:
These are two distinct validation points, and it's important to understand the scope of existing tests.

Line 192 validates multi-region configurations, while line 177 validates multi-rack configurations.

Tests for multi-rack only (not multi DC) configurations currently exist. However, these tests may not fully cover scenarios where the 'DC of loader' cannot be located or is unavailable. This means that if the 'DC of loader' is not found for any reason, the system's behaviour in multi-rack configurations might not be validated.